pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -768,10 +788,62 @@ func (r *ReconcileBareMetalHost) actionManageReady(prov provisioner.Provisioner,
 	return r.manageHostPower(prov, info)
 }
 
-func (r *ReconcileBareMetalHost) saveStatus(host *metal3v1alpha1.BareMetalHost) error {
+func (r *ReconcileBareMetalHost) saveHostStatus(host *metal3v1alpha1.BareMetalHost) error {
 	t := metav1.Now()
 	host.Status.LastUpdated = &t
-	return r.client.Status().Update(context.TODO(), host)
+
+	if err := r.saveHostAnnotation(host); err != nil {
+		return err
+	}
+
+	//Refetch host again
+	obj := host.Status.DeepCopy()
+	err := r.client.Get(context.TODO(),
+		client.ObjectKey{
+			Name:      host.Name,
+			Namespace: host.Namespace,
+		},
+		host,
+	)
+	if err != nil {
+		return errors.Wrap(err, "Failed to update Status annotation")
+	}
+	host.Status = *obj
+	errStatus := r.client.Status().Update(context.TODO(), host)
+	return errStatus
+}
+
+func (r *ReconcileBareMetalHost) saveHostAnnotation(host *metal3v1alpha1.BareMetalHost) error {
+	//Repopulate annotation again
+	objStatus := r.getHostStatusFromAnnotation(host)
+	if objStatus != nil {
+		objStatus.LastUpdated = host.Status.LastUpdated
+	}
+
+	if !reflect.DeepEqual(host.Status, objStatus) {
```

Comment:
The usual pattern in go is to prefer early exits so that the "happy path" through a function is not inside a conditional. So, if you reverse this test then when the values are equal this function should return nil. Then lines 824-833 can move outside of a conditional and be the main body of the function.

Reply:
Done